Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -613,6 +790,25 @@ extension BidirectionalCollection where SubSequence == Substring {
         maxSplits: Int = Int.max,
         omittingEmptySubsequences: Bool = true
     ) -> some Collection<Substring>
+    
+    /// Returns the longest possible subsequences of the collection, in order,
+    /// around subsequence that match the given separator regex.
+    ///
+    /// - Parameters:
+    ///   - maxSplits: The maximum number of times to split the collection, 
+    ///     or one less than the number of subsequences to return.
+    ///   - omittingEmptySubsequences: If `false`, an empty subsequence is 
+    ///     returned in the result for each consecutive pair of matches
+    ///     and for each match at the start or end of the collection. If
+    ///     `true`, only nonempty subsequences are returned.
+    ///   - separator: A closure to produce a `RegexComponent` to be split upon.
+    /// - Returns: A collection of substrings, split from this collection's
+    ///   elements.
+    public func split(
+        maxSplits: Int = Int.max,
+        omittingEmptySubsequences: Bool = true,
+        @RegexComponentBuilder _ separator: () -> some RegexComponent
```

Comment:
Should we keep this argument label, so that when the user is not using a trailing closure they will be using a `separator:` label to clarify the argument?
```suggestion
        @RegexComponentBuilder separator: () -> some RegexComponent
```

Reply:
I changed the argument labels of the builder functions to match their non-builder counterparts.